impl/src/main/java/org/jboss/weld/util/Beans.java

```diff
@@ -647,6 +652,13 @@ public static boolean hasBuiltinScope(Bean<?> bean) {
                 || ConversationScoped.class.equals(bean.getScope()) || Dependent.class.equals(bean.getScope());
     }
 
+    public static boolean hasBuiltinScope(AnnotatedType<?> annotatedType) {
+        Set<Annotation> annotations = annotatedType.getAnnotations();
```

Comment:
I would use `AnnotatedType.isAnnotationPresent()` instead (so that those literals are not needed).